webui/src/lib/onboarding/repoOnboardingService.ts

```diff
@@ -0,0 +1,119 @@
+import { OnBoardingSteps } from "./types";
+import { NavigateFunction } from "react-router-dom";
+import { branches, retention, branchProtectionRules, objects } from "../api";
+
+const IMPORT_BRANCH_REGEX = /^_(.*)_imported$/;
+const HOOKS_PATH = "_lakefs_actions/";
+const SAMPLE_REPO_FILE_NAME = "lakes.parquet";
+
+// generate a random file name to avoid collisions for the upload preflight check
+const generateRandomFileName = (): string => {
+  const randomString = Math.random().toString(36).substring(2, 15);
+  return `sample_${randomString}.yaml`;
+};
+
+export const isSampleRepo = async (
+  repo: string,
+  defaultBranch = "main"
+): Promise<boolean> => {
+  const objectList = await objects.list(repo, defaultBranch, "");
+  return objectList.results.some(
+    // when the API client is typed, we can remove the eslint-disable
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    (object: any) => object?.path === SAMPLE_REPO_FILE_NAME
+  );
+};
+
+export const canUseRepoOnboarding = async (
+  repoId: string,
+  defaultBranch = "main"
+): Promise<boolean> => {
+  const promises = [];
+  promises.push(retention.setGCPolicyPreflight(repoId));
+  promises.push(branchProtectionRules.createRulePreflight(repoId));
+  promises.push(
+    objects.list(
+      repoId,
+      defaultBranch,
+      `${HOOKS_PATH}${generateRandomFileName()}`
+    )
+  );
+  const results = await Promise.allSettled(promises);
```

Comment:
This waits for all 3 calls to finish even after it knows the result.  The long-term case is of an onboarded user, and this code is slower for all calls once the user is onboarded.

We could `Promise.all` here, e.g.
```js
return Promise.all(results).then(_.some).catch(false);
```
It will often be faster, and never slower.

Reply:
You're right. We should use `all` and not `allSettled` in this case.

Reply:
Done.